File: travel-frontend/src/types/booking.ts
// ==================== src/types/booking.ts ====================
export interface Booking {
  id: number;
  userId: number;
  userName: string;
  travelPackageId: number;
  packageTitle: string;
  bookingDate: string;
  numberOfTravelers: number;
  totalPrice: number;
  status: string;
  specialRequests?: string;
  createdAt: string;
}

export interface CreateBookingRequest {
  travelPackageId: number;
  numberOfTravelers: number;
  specialRequests?: string;
}

File: src/TravelManagement.API/Models/Entities/Booking.cs
// ==================== Booking.cs ====================
namespace TravelManagement.API.Models.Entities;

public class Booking : BaseEntity
{
    public int UserId { get; set; }
    public int TravelPackageId { get; set; }
    public DateTime BookingDate { get; set; } = DateTime.UtcNow;
    public int NumberOfTravelers { get; set; }
    public decimal TotalPrice { get; set; }
    public string Status { get; set; } = "Pending"; // Pending, Confirmed, Cancelled
    public string? SpecialRequests { get; set; }

    // Navigation properties
    public User User { get; set; } = null!;
    public TravelPackage TravelPackage { get; set; } = null!;
}